lib/npg_qc/autoqc/checks/pulldown_metrics.pm

```diff
@@ -56,28 +57,51 @@ has 'max_java_heap_size' => (
     default => $MAX_JAVA_HEAP_SIZE,
 );
 
-has 'picard_jar_path' => (
+has 'picard_module' => (
     is      => 'ro',
-    isa     => 'NpgCommonResolvedPathJarFile',
-    coerce  => 1,
-    default => $PICARD_JAR_NAME,
+    isa     => 'Str',
+    default => $PICARD_COMMAND,
 );
 
-has 'picard_command' => (
-    is         => 'ro',
-    isa        => 'Str',
+# TODO: This needs to be factored out of here, insert_size and alignment_filter_metrics
+has 'reference' => (
+    is => 'ro',
+    isa => 'Str',
     lazy_build => 1,
 );
 
-sub _build_picard_command {
+sub _build_reference {
     my $self = shift;
-    my $command = $self->java_cmd . sprintf q[ -Xmx%s -jar %s VALIDATION_STRINGENCY=SILENT BAIT_INTERVALS=%s TARGET_INTERVALS=%s INPUT=%s OUTPUT=/dev/stdout],
-        $self->max_java_heap_size,
-        $self->picard_jar_path,
-        $self->bait_intervals_path,
-        $self->target_intervals_path,
-        $self->input_files->[0];
-    return $command;
+    my $ref;
+    try {
+        $ref = pop @{$self->refs()};
+    }
+    catch {
+        $self->result->add_comment(
+            q[Error: binary reference cannot be retrieved; cannot run insert size check. ] . $_
+        );
+    };
+    if (!$ref) {
+        $self->result->add_comment('Failed to retrieve reference');
```

Comment:
Error here with a clear message about a failure to get a FASTA file path, otherwise later we will get much more difficult to interpret errors.